core/app/models/spree/shipment.rb

```diff
@@ -430,5 +518,28 @@ def ensure_can_destroy
         throw :abort
       end
     end
+
+    def store_state_change(previous_state, new_state)
+      state_changes.create!(
+        previous_state: previous_state,
+        next_state:     new_state,
+        name:           'shipment'
+      )
+    end
+
+    def change_state!(new_state)
+      previous_state = state
+      if new_state != previous_state
+        update_attributes(
+          state: new_state,
+          updated_at: Time.current
+        )
```

Comment:
ActiveRecord already sets `updated_at` for us.

Also, the bang at the end of the method name indicates that this method raises if the record couldn't be saved. We should use `update!` here.

```rb
update!(state: new_state)
```

Reply:
We should consider to move the methods `change_state!` and `store_state_change` into a module we share with all models that have `state` changes.